### ISSUE TITLE
codec: separate mimetype for encoding and decoding

### ISSUE THREAD

**Issue by dolmen** [dolanor/rip#7]
This is necessary to remove hack for HTML encoding which is is not symmetrical.

**Comment by dolanor** [comment]
Do you mean because HTML codec has only encoding and HTML Forms codec only has decoding?

**Comment by dolmen** [comment]
Yes. `AcceptEncoder` has hardcoded references to that specific codec.

https://github.com/dolanor/rip/blob/379a63bc43960312259437251ace0f4c1e1a8287/encoding/encoding.go#L90

**Comment by dolanor** [comment]
Yeah, there is some experimentation of content negociation still all over the place once I played with HTML page and forms. It bled a little more than what I wanted.
One day when I'm fresh and a clear mind, I'll refactor all of this, because it is messy.

**Comment by jub0bs** [comment]
Perhaps relevant: https://github.com/golang/go/issues/19307

**Comment by dolanor** [comment]
I don't think it's relevant to the discussion here, but it is relevant to my interest as I will be able to get rid of my implementation of content negotiation 😁 

https://github.com/dolanor/rip/blob/main/header.go
https://github.com/dolanor/rip/blob/main/header_test.go

I'm tempted to run the test on my implementation to see how bad I fared 😛 

Thanks for letting me now of this proposal. (for my future self => [CL](https://go-review.googlesource.com/c/net/+/642956/7) )

**Comment by dolanor** [comment]
I ran the test, and I was not so bad. Basically not handling the languages and charset encoding, but it was not the purpose of my function.
And then, not handling the offer order as a basis for selection.

I'm gonna create an issue to use the stdlib once it's merged.
